src/node/gov/handlers/proposals.h

```diff
@@ -250,26 +253,31 @@ namespace ccf::gov::endpoints
             else
             {
               proposal_info.state = ProposalState::FAILED;
-              failure = ccf::jsgov::Failure{
+              proposal_info.failure = ccf::jsgov::Failure{
                 fmt::format(
                   "resolve() returned invalid status value: \"{}\"", status),
                 std::nullopt // No trace
               };
             }
           }
+
+          // Ensure resolved proposal_info is visible in the KV
+          proposal_info_handle->put(proposal_id, proposal_info);
         }
 
         if (proposal_info.state != ProposalState::OPEN)
         {
           remove_all_other_non_open_proposals(tx, proposal_id);
-          final_votes = std::unordered_map<ccf::MemberId, bool>();
-          for (auto& [mid, vote] : votes)
-          {
-            final_votes.value()[mid] = vote;
-          }
 
           if (proposal_info.state == ProposalState::ACCEPTED)
           {
+            // Write now-permanent values back to proposal_state, and into the
```

Comment:
I think we want to do this for REJECTED also, this is about final vs OPEN, not ACCEPTED.

Reply:
Plus a question mark for withdrawn proposals. I claim it's not particularly useful to evaluate final votes and store them in that case, but it may be arguable.